### ISSUE TITLE
extra points for survival

### ISSUE THREAD

**Issue by bretonium** [ResamVi/infclass-stats#1]
Survival is hard. Currently the percentage of human survival jumps from 5 to 15%. But the stats do no motivate people to survive. It is more rewarding to get a lot of points and die than survive. Which is not good, giving that survival (human win) is the point of the game.

I suggest to motivate players more for surviving. More points should be given for survival and, maybe, points should be deducted for dying

**Comment by ResamVi** [comment]
```
But the stats do no motivate people to survive.
```
Short Answer: The stats rewards players that score above average in kills and survival so it does motivate people. It is sort of self-regulating: If all players take no focus in surviving -> the average is low -> Being twice/thrice/multiple times as good as the average is easier -> yields greater score results


Long Answer:
# Let's take your assumption and do the math: "Players focus on kills rather than surviving"

Math from here: https://github.com/ResamVi/infclass-stats/blob/master/operations/processing.go#L194



Stats taken on `2019-11-26T16:50:49+0000`

Average Kills of all Players:

![image](https://user-images.githubusercontent.com/6261556/69654360-f02db700-1074-11ea-91dd-268fbd23288f.png)

Average Survival of all Players (total Survival / total Players): 0.1

![image](https://user-images.githubusercontent.com/6261556/69654446-1c493800-1075-11ea-98f0-dc34423dad17.png)


**The perfectly average Player**
- Has gained a total of 107 kills
- Survived once

```
1000 * log2(1 + (107/106.2747)) + 1000 * log2(1 + (1/0.8213))
```

Puts him at a score of **2153**

**The perfectly average Player survives once more**
- He played medic and thus stays at a total of 107 kills
- Survived twice
- I'll throw you a bone: the average increases more than a significant portion (meaning surviving is less impactful on your score, so he would be gaining less than if it had stayed the same)

```
1000 * log2(1 + (107/106.2747)) + 1000 * log2(1 + (2 / 0.8313))
```
Puts him at a score of **2785**. A Net score of **+632**

**A different perfectly average Player that didn't survive but went for kills**
- Has gained an additional 30 kills to reach a total of 137 kills
- Survived once
- The average of kills stay the same (So more kills should actually be more impactful)

```
1000 * log2(1 + (137/106.2747)) + 1000 * log2(1 + (1 / 0.8313))
```
Puts him at a score of **2334**. A Net score of **+181**

**He would have to get an additional 119 (total of 226) to go even with the surviver**

I'd say if you can sustain this kind of performance you are free to do so.


So in conclusion the current scoring system has a lot of room of improvement (playing more = more score)
but this is not exactly one of its pain points I'd say.
